### PR TITLE
feat: discover ember service dependency information from a file

### PR DESCRIPTION
**Pull request by wondersloth**
This is work to support feature #390.
This PR addresses parsing a single source file and finding the usages of `@ember/service`. 
It returns an array of service meta data.

```
type EmberInferredServiceDependency = {
  addonName?: string;
  serviceName: string;
};
```

This implementation should not try resolve the services just parse and extract meaningful meta data.

### CODE REVIEW

**Comment by lynchbomb** [comment]
Since this implementation is specific to Ember shouldn't we have it live in `migration-graph-ember` and not `migration-graph`?

**Comment by wondersloth** [comment]
> Since this implementation is specific to Ember shouldn't we have it live in `migration-graph-ember` and not `migration-graph`?

- [ ] Move to `migration-graph-ember`

I can move it over there now. My other work is sitting in `migration-graph` I hadn't thought about organization yet because I don't like having to "prepare" across packages.

File: packages/migration-graph-ember/src/utils/discover-ember-service-dependencies.ts [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+  const maybeImportDeclaration = findImportDeclarationsBySource(parsed.body, '@ember/service');
+
+  if (!maybeImportDeclaration) {
+    DEBUG_CALLBACK('No import declaration found');
+    return EMPTY_RESULT;
+  }
+
+  const importDeclaration = maybeImportDeclaration as ImportDeclaration;
+
+  // In the case of re-assignment from ineject as service, we walk the specifiers
```
Comment (chadhietala) [inline packages/migration-graph-ember/src/utils/discover-ember-service-dependencies.ts:46]:
```suggestion
In the case of re-assignment from inject as service, we walk the specifiers
```

File: packages/migration-graph-ember/src/utils/discover-ember-service-dependencies.ts [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    // Check if the ExportDefaultDeclaration has a ClassExpression
+    if (edd.decl.type === 'ClassExpression') {
+      foundClasses.push(edd.decl as ClassExpression);
+    } else {
+      DEBUG_CALLBACK('Default export does not contain a ClassExpression');
+    }
+  }
+
+  if (foundClasses.length < 1) {
+    throw new Error('Unexpected Error: No classes found despite having imported @ember/service.');
```
Comment (chadhietala) [inline packages/migration-graph-ember/src/utils/discover-ember-service-dependencies.ts:84]: I don't think we want to throw in the case. Probably should just return out with an empty result.

**Review by chadhietala — CHANGES_REQUESTED** [review]
Overall looks good to me. Just a couple things that I think need to be fixed.

**Comment by chadhietala** [comment]
Please make sure to migrate the ember graph stuff to use SWC instead of babel

**Comment by lynchbomb** [comment]
https://github.com/rehearsal-js/rehearsal-js/issues/365

**Review by lynchbomb — APPROVED** [review]

**Pull request merged**